src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/ObjectPools/PooledDisposer.NonCopyable.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.PooledObjects;
+
+[NonCopyable]
```

Comment:
this is because NonCopyable is only a thing in Workspaces, not in the core compiler?

Reply:
https://github.com/dotnet/roslyn/pull/67198#discussion_r1127032764

Reply:
yeah, it's that hte attribute it defined at a higher layer.  i'm ok with moving it down if you guys are.  it's *fantastic*.  Really valuable for these sorts of cases where you want the power and value of a struct, without some of its annoying downsides.   Would love a language feature for this case.  But this serves as a useful alternative until we get there.